### PR TITLE
Test APCu __construct() to return RuntimeException if APCu extension not installed

### PR DESCRIPTION
**Pull request by Laurent3170**

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace Doctrine\Tests\Common\Cache;
+
+use Doctrine\Common\Cache\ApcuCache;
+use Doctrine\Common\Cache\CacheProvider;
+
+/**
+ * Tests if APCu extension is installed
+ */
+class ApcuExtensionTest extends CacheTest
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: `ApcuCacheRejectsMissingApcuExtensionTest`

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Doctrine\Tests\Common\Cache;
+
+use Doctrine\Common\Cache\ApcuCache;
+use Doctrine\Common\Cache\CacheProvider;
+
+/**
+ * Tests if APCu extension is installed
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: Comment can be dropped after a rename

Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: Add an `@covers \Doctrine\Common\Cache\ApcuCache`

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php (hunk 1 above)
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: Please simply extend `\PHPUnit_Framework_TestCase`

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+class ApcuExtensionTest extends CacheTest
+{
+    protected function setUp() : void
+    {
+        if (function_exists('apcu_fetch')) {
+            $this->markTestSkipped('APCu is actually enabled');
+        }
+    }
+
+    protected function _getCacheDriver() : CacheProvider
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: Remove - instantiate directly in the test method

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        return new ApcuCache();
+    }
+
+    public function testApcuExtension() : void
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: `testApcuExtensionIsRequired() : void`

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        if (! function_exists('apcu_fetch')) {
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: Not needed: you already check this in `setUp`

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            $this->expectException(RuntimeException::ApcuCache);
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: `$this->expectException(RuntimeException::class)`

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+            new ApcuCache();
+        } else {
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: `else` is a keyword that you should strive to get rid of in your codebase

**Comment by Ocramius** [comment]
Closing - please continue in https://github.com/doctrine/cache/pull/225 - consider the review comments above when incorporating the test

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php (hunk 5 above)
Comment (Laurent3170) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: But, if not needed then the method `testApcuExtension()` (becoming `testApcuExtensionIsRequired()`) will be empty and the test will only contain `setUp()` method. Is it right?

File: tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php (hunk 7 above)
Comment (Laurent3170) [inline tests/Doctrine/Tests/Common/Cache/ApcuExtensionTest.php]: I was thinking we should always give an else

**Pull request closed without merging**